### ISSUE TITLE
Bug: ThrowStatement's argument should be Expression

### ISSUE THREAD

**Issue by ericmorand** [typescript-eslint/typescript-eslint#9610]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Issue

According to the ECMAScript specification, ThrowStatement is required to have an expression as argument:

https://tc39.es/ecma262/#sec-throw-statement

But in ast-spec, the argument is exposed as a statement or null:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/ast-spec/src/statement/ThrowStatement/spec.ts

```
import type { AST_NODE_TYPES } from '../../ast-node-types';
import type { BaseNode } from '../../base/BaseNode';
import type { TSAsExpression } from '../../expression/TSAsExpression/spec';
import type { Statement } from '../../unions/Statement';

export interface ThrowStatement extends BaseNode {
  type: AST_NODE_TYPES.ThrowStatement;
  argument: Statement | TSAsExpression | null;
}
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types` | `7.16.1` |

**Comment by Josh-Cena** [comment]
Hi! Although `throw;` is invalid syntax, the TypeScript parser we use under the hood accepts it and emits an AST for it (although also reporting a semantic error). Our AST spec describes all ASTs that may be emitted by our parser, not just those that represent well-formed programs. To fix this, we need to add a grammar check to our own parser before emitting the AST.

**Comment by Josh-Cena** [comment]
The type for `ThrowStatement` doesn't make sense. `argument` should be an expression, not a statement.

**Comment by ericmorand** [comment]
@Josh-Cena it is a fair point (the edited part about the parser being tolerant). I'm writing a code coverage tool and I think it may benefit from the flexibility that your parser provides: after all, one could totally write an incorrect JS or TS code `throw;` and the coverage tool would need to not crash because of the invalid syntax. So, yeah, it makes sense.

About the expression part, I agree that it should be an expression. Not a big deal but worth fixing.

**Comment by Josh-Cena** [comment]
`argument` being `Statement` is actually more severe than it containing `null`—this means it can't actually derive any valid AST because `Statement` and `Expression` have no overlap, whereas an additional `| null` is sound but just useless!

**Comment by ericmorand** [comment]
> whereas an additional | null is sound but just useless!

The only issue that it creates is that it either force developers to add an uncoverable `if` to their code, or to force type argument as not-null:

```ts
case AST_NODE_TYPES.ThrowStatement: {
      if (node.argument) { // barely coverable branch
          handleNodeWithPassedRangeCoverages(node.argument);
      }
      
      break;
}
```

```ts
case AST_NODE_TYPES.ThrowStatement: {
      // @see https://github.com/typescript-eslint/typescript-eslint/issues/9610
      handleNodeWithPassedRangeCoverages(node.argument!); // better but not very elegant

      break;
}
```

**Comment by bradzacher** [comment]
It's really funny because you can trace this all the way back to when the types were first added in #156. Nobody has noticed this was wildly incorrect since the type was implemented 🤣.
